packages/client/src/pages/attendance/index.tsx

```diff
@@ -74,10 +82,10 @@ const AttendancePage: React.FC = () => {
     { collection: OrgSubCollection.SlotsByDay },
   ]);
 
+  const calendarNavProps = useDate();
   const [view, setView] = useState<keyof typeof viewsLookup>(Views.ByDay);
-  const AttendanceView = viewsLookup[view];
+  const AttendanceView = viewsLookup[view]([calendarNavProps]);
```

Comment:
I don't really understand why the `resetKeys` is necessary (my guess is it has something to do with the memoization).

However...this is React, not svelte. By that I mean...every time a rerender happens, this function will be called, to retrieve the view function. This, in turn, means that the returned JSX will be fully rerendered (as the component has changed). This is something you pick up from years of using React, I would like to not think about it and I wouldn't want anybody else to have to think about it, so let me summarise:

It would be better to simply pass `reserKeys` to the view component, instead of having the component be a higher order function, returning a component, like so:

```jsx
[Views.ByDay]: ({resetKeys}: {resetKeys: Array<any>}) =>
    (
      <LayoutContent>
        <ErrorBoundary resetKeys={[resetKeys]}>
          <ByDayView />
        </ErrorBoundary>
      </LayoutContent>
    ),
```

And then you simply do:

```jsx
<Layout>
  <Calendar {...calendarProps} />
  <AttendanceView resetKeys={resetKeys} />
</Layout>
```

Reply:
Going back to this...what does the `resetKeys` do? I removed it completely and didn't observe any different behaviour...

Reply:
ResetKeys trigger a reset of the error boundary when they change. When you removed it did the error boundary disappear and non-error view get rendered when you switched between dates?